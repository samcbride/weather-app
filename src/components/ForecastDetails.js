import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

const ForecastDetails = ({ forecast }) => {
  const { date, humidity, temperature, wind } = forecast;
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        {moment(date).format("ddd do MMM")}
      </div>
      <div className="forecast-details__max-temperature">
        Max Temperature:
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__min-temperature">
        Min temperature:
        {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__humidity">
        Humidity:
        {humidity}
      </div>
      <div className="forecast-details__wind-speed">
        Wind speed:
        {wind.speed}
      </div>
      <div className="forecast-details__wind-direction">
        Wind direction:
        {wind.direction}
      </div>
    </div>
  );
};

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    humidity: PropTypes.number.isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
