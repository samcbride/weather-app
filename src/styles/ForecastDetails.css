.forecast-details {
  text-align: center;
  padding: 2em;
}

.forecast-details__date {
  font-size: 30px;
}
