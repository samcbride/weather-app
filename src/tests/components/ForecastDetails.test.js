import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      max: 22,
      min: 12,
    },
    humidity: 80,
    wind: {
      speed: 60,
      direction: "sw",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
