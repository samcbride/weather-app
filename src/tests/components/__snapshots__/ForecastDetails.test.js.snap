// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu 4th Jan
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      Max Temperature:22°C
    </div>
    <div
      class="forecast-details__min-temperature"
    >
      Min temperature:12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity:80
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      Wind speed:60
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      Wind direction:sw
    </div>
  </div>
</DocumentFragment>
`;
